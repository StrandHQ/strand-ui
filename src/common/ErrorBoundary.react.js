import {Component} from 'react';
import PropTypes from 'prop-types';

class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = {error: null};
  }

  componentDidCatch(error, errorInfo) {
    if (Raven.isSetup()) {
      this.setState({error});
      Raven.captureException(error, {extra: errorInfo});
    }
  }

  render() {
    // TODO could add a nice "there was an error, help us out" kind of screen here
    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.object,
};

export default ErrorBoundary;
