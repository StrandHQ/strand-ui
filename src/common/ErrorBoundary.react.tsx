import {Component} from 'react';
import * as Raven from 'raven-js';

interface PropTypes {
  children: object,
}

class ErrorBoundary extends Component<PropTypes> {
  constructor(props) {
    super(props);
    this.state = {error: null};
  }

  componentDidCatch(error, errorInfo) {
    if (Raven.isSetup()) {
      this.setState({error});
      Raven.captureException(error, {extra: errorInfo});
    }
  }

  render() {
    // TODO could add a nice "there was an error, help us out" kind of screen here
    return this.props.children;
  }
}

export default ErrorBoundary;
