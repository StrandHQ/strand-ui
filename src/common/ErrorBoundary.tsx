import {Component} from 'react';
import * as Raven from 'raven-js';

interface PropTypes {
  children: object,
}

class ErrorBoundary extends Component<PropTypes> {
  constructor(props: PropTypes) {
    super(props);
    this.state = {error: null};
  }

  componentDidCatch(error: Error, errorInfo: object) {
    if (Raven.isSetup()) {
      this.setState({error});
      Raven.captureException(error, {extra: errorInfo});
    }
  }

  render() {
    // TODO could add a nice "there was an error, help us out" kind of screen here
    return this.props.children;
  }
}

export default ErrorBoundary;
