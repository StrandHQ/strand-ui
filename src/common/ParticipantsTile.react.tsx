import * as React from 'react';
import {Component} from 'react';
import Divider from 'material-ui/Divider';
import List from 'material-ui/List';
import UserListItem from './UserListItem.react';
import {filter} from 'lodash';
import {GetTopicUserFragment} from '../../schema/graphql-types';

interface PropTypes {
  originalPoster: GetTopicUserFragment
  // TODO [UI-50] No nullable arrays
  participants: (GetTopicUserFragment | null)[] | null
}

class ParticipantsTile extends Component<PropTypes> {
  nonOriginalPosterParticipants(participants: (GetTopicUserFragment | null)[]) {
    return filter(participants, participant => this.props.originalPoster.id !== participant!.id);
  }

  render() {
    const participants = this.props.participants || [];
    const otherParticipants = this.nonOriginalPosterParticipants(participants);
    return (
      <List dense>
        <UserListItem user={this.props.originalPoster} isOriginalPoster/>
        <Divider inset/>
        {otherParticipants.map(participant => (
          <div key={`participant_${participant!.id}`}>
            <Divider inset/>
            <UserListItem user={participant!}/>
          </div>
        ))}
      </List>
    );
  }
}

export default ParticipantsTile;
