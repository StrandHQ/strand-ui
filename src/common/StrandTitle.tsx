import * as React from 'react';
import {Component} from 'react';
import Typography from 'material-ui/Typography';
import {Link} from 'react-router-dom';

interface PropTypes {
  title: string | null
  strandId: string
  center?: boolean
}

class StrandTitle extends Component<PropTypes> {
  render() {
    return (
      <Typography
        variant='title'
        align={this.props.center ? 'center' : undefined}>
        <Link to={`/strands/${this.props.strandId}`} style={{textDecoration: 'unset'}}>
          {this.props.title || '[No Title]'}
        </Link>
      </Typography>
    );
  }
}

export default StrandTitle;
