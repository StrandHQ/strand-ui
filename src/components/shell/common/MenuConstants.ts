interface NavigationPaths {
  [label: string] : string,
}

export const navigationLabelToPath: NavigationPaths = {
  Install: '/install',
  Strands: '/strands',
};

export default {navigationLabelToPath};
