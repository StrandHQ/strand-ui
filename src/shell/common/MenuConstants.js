export const navigationLabelToPath = {
  'Install': '/install',
};

export default {navigationLabelToPath};
