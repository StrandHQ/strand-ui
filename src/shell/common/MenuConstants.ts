interface NavigationPaths {
  [label: string] : string,
}

export const navigationLabelToPath: NavigationPaths = {
  Install: '/install',
  Topics: '/topics',
};

export default {navigationLabelToPath};
