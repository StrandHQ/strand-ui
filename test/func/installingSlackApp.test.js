import {navigationLabelToPath} from 'src/shell/common/MenuConstants';
import {mountApplication} from 'test/helper/applicationMock';
import InstallationStatus from 'src/install/InstallationStatus.react';

function flushPromises() {
  return new Promise(resolve => setImmediate(resolve));
}

describe('installing slack app', () => {
  it('sends the user to Slack when the user navigates to the page and clicks the button', async () => {
    const wrapper = mountApplication('/');

    // Navigate to install page
    wrapper.find(`Button[id="${navigationLabelToPath.Install}-button"]`).prop('onClick')();

    // Assert that the install button goes to Slack
    const anchorHref = wrapper.find('a[id="add-to-slack-button"]').prop('href');
    expect(anchorHref).toContain('slack.com/oauth');
    expect(anchorHref).toContain(process.env.SLACK_CLIENT_ID);
  });

  it('shows a success message when a user successfully installs the app', async () => {
    const mockSlackCode = '12345';
    const graphqlMocks = {
      AttemptSlackInstallationMutation: (_, info) => {
        expect(info.input.code).toEqual(mockSlackCode); // ensuring the code is pulled from the URL
        return {}
      },
    };
    const wrapper = mountApplication(`${navigationLabelToPath.Install}?code=${mockSlackCode}`, {graphqlMocks});
    await flushPromises(); // wait for GraphQL call to complete
    wrapper.update();

    const installationStatusComponent = wrapper.find(InstallationStatus);
    expect(installationStatusComponent).toHaveLength(1);
    expect(installationStatusComponent).toMatchSnapshot();
  });

  it('shows a failure message when we fail to install the app', async () => {
    const graphqlMocks = {
      AttemptSlackInstallationMutation: () => {
        throw 'This error message should be displayed on UI'
      },
    };
    const wrapper = mountApplication(`${navigationLabelToPath.Install}?code=12345`, {graphqlMocks});
    await flushPromises(); // wait for GraphQL call to complete
    wrapper.update();

    const installationStatusComponent = wrapper.find(InstallationStatus);
    expect(installationStatusComponent).toHaveLength(1);
    expect(installationStatusComponent).toMatchSnapshot();
  })
});
