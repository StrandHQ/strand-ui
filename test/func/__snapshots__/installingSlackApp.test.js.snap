// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`installing slack app shows a failure message when we fail to install the app 1`] = `
<InstallationStatus
  errors={
    Array [
      "This error message should be displayed on UI",
    ]
  }
  installingSlackApplication={false}
  successInstallationSlackApplication={false}
>
  <WithStyles(Typography)
    style={
      Object {
        "color": "red",
      }
    }
    variant="subheading"
  >
    <Typography
      align="inherit"
      classes={
        Object {
          "alignCenter": "MuiTypography-alignCenter-165",
          "alignJustify": "MuiTypography-alignJustify-167",
          "alignLeft": "MuiTypography-alignLeft-164",
          "alignRight": "MuiTypography-alignRight-166",
          "body1": "MuiTypography-body1-161",
          "body2": "MuiTypography-body2-160",
          "button": "MuiTypography-button-163",
          "caption": "MuiTypography-caption-162",
          "colorError": "MuiTypography-colorError-175",
          "colorInherit": "MuiTypography-colorInherit-171",
          "colorPrimary": "MuiTypography-colorPrimary-172",
          "colorSecondary": "MuiTypography-colorSecondary-173",
          "colorTextSecondary": "MuiTypography-colorTextSecondary-174",
          "display1": "MuiTypography-display1-156",
          "display2": "MuiTypography-display2-155",
          "display3": "MuiTypography-display3-154",
          "display4": "MuiTypography-display4-153",
          "gutterBottom": "MuiTypography-gutterBottom-169",
          "headline": "MuiTypography-headline-157",
          "noWrap": "MuiTypography-noWrap-168",
          "paragraph": "MuiTypography-paragraph-170",
          "root": "MuiTypography-root-152",
          "subheading": "MuiTypography-subheading-159",
          "title": "MuiTypography-title-158",
        }
      }
      color="default"
      gutterBottom={false}
      headlineMapping={
        Object {
          "body1": "p",
          "body2": "aside",
          "display1": "h1",
          "display2": "h1",
          "display3": "h1",
          "display4": "h1",
          "headline": "h1",
          "subheading": "h3",
          "title": "h2",
        }
      }
      noWrap={false}
      paragraph={false}
      style={
        Object {
          "color": "red",
        }
      }
      variant="subheading"
    >
      <h3
        className="MuiTypography-root-152 MuiTypography-subheading-159"
        style={
          Object {
            "color": "red",
          }
        }
      >
        Failed to install.
        <br />
        Errors:Â 
        This error message should be displayed on UI
      </h3>
    </Typography>
  </WithStyles(Typography)>
</InstallationStatus>
`;

exports[`installing slack app shows a success message when a user successfully installs the app 1`] = `
<InstallationStatus
  errors={Array []}
  installingSlackApplication={false}
  successInstallationSlackApplication={true}
>
  <WithStyles(Typography)
    style={
      Object {
        "color": "green",
      }
    }
    variant="subheading"
  >
    <Typography
      align="inherit"
      classes={
        Object {
          "alignCenter": "MuiTypography-alignCenter-165",
          "alignJustify": "MuiTypography-alignJustify-167",
          "alignLeft": "MuiTypography-alignLeft-164",
          "alignRight": "MuiTypography-alignRight-166",
          "body1": "MuiTypography-body1-161",
          "body2": "MuiTypography-body2-160",
          "button": "MuiTypography-button-163",
          "caption": "MuiTypography-caption-162",
          "colorError": "MuiTypography-colorError-175",
          "colorInherit": "MuiTypography-colorInherit-171",
          "colorPrimary": "MuiTypography-colorPrimary-172",
          "colorSecondary": "MuiTypography-colorSecondary-173",
          "colorTextSecondary": "MuiTypography-colorTextSecondary-174",
          "display1": "MuiTypography-display1-156",
          "display2": "MuiTypography-display2-155",
          "display3": "MuiTypography-display3-154",
          "display4": "MuiTypography-display4-153",
          "gutterBottom": "MuiTypography-gutterBottom-169",
          "headline": "MuiTypography-headline-157",
          "noWrap": "MuiTypography-noWrap-168",
          "paragraph": "MuiTypography-paragraph-170",
          "root": "MuiTypography-root-152",
          "subheading": "MuiTypography-subheading-159",
          "title": "MuiTypography-title-158",
        }
      }
      color="default"
      gutterBottom={false}
      headlineMapping={
        Object {
          "body1": "p",
          "body2": "aside",
          "display1": "h1",
          "display2": "h1",
          "display3": "h1",
          "display4": "h1",
          "headline": "h1",
          "subheading": "h3",
          "title": "h2",
        }
      }
      noWrap={false}
      paragraph={false}
      style={
        Object {
          "color": "green",
        }
      }
      variant="subheading"
    >
      <h3
        className="MuiTypography-root-152 MuiTypography-subheading-159"
        style={
          Object {
            "color": "green",
          }
        }
      >
        Successfully installed! The Strand bot should have DM'd you with setup instructions.
      </h3>
    </Typography>
  </WithStyles(Typography)>
</InstallationStatus>
`;
